mslib/msui/mpl_qtwidget.py

```diff
@@ -1296,98 +1452,27 @@ def redraw_map(self, kwargs_update=None):
-        """
-        logging.debug("plotting image..")
-        self.wms_image = self.map.imshow(img, interpolation="nearest", origin=PIL_IMAGE_ORIGIN)
-        # NOTE: imshow always draws the images to the lowest z-level of the
-        # plot.
-        # See these mailing list entries:
-        # http://www.mail-archive.com/matplotlib-devel@lists.sourceforge.net/msg05955.html
-        # http://old.nabble.com/Re%3A--Matplotlib-users--imshow-zorder-tt19047314.html#a19047314
-        #
-        # Question: Is this an issue for us or do we always want the images in the back
-        # anyhow? At least we need to remove filled continents here.
-        # self.map.set_fillcontinents_visible(False)
-        # ** UPDATE 2011/01/14 ** seems to work with version 1.0!
-        logging.debug("done.")
+        self.myfig.draw_image(img)
 
     def draw_legend(self, img):
         """Draw the legend graphics img on the current plot.
-
         Adds new axes to the plot that accomodate the legend.
         """
-        # If the method is called with a "None" image, the current legend
-        # graphic should be removed (if one exists).
-        if self.legimg is not None:
-            logging.debug("removing image %s", self.legimg)
-            self.legimg.remove()
-            self.legimg = None
-
-        if img is not None:
-            # The size of the legend axes needs to be given in relative figure
-            # coordinates. To determine those from the legend graphics size in
-            # pixels, we need to determine the size of the currently displayed
-            # figure in pixels.
-            figsize_px = self.fig.get_size_inches() * self.fig.get_dpi()
-            ax_extent_x = float(img.size[0]) / figsize_px[0]
-            ax_extent_y = float(img.size[1]) / figsize_px[1]
-
-            # If no legend axes have been created, do so now.
-            if self.legax is None:
-                # Main axes instance of mplwidget has zorder 99.
-                self.legax = self.fig.add_axes([1 - ax_extent_x, 0.01, ax_extent_x, ax_extent_y],
-                                               frameon=False,
-                                               xticks=[], yticks=[],
-                                               label="ax2", zorder=0)
-                self.legax.patch.set_facecolor("None")
-                # self.legax.set_yticklabels("ax2", fontsize=32, minor=False)
-                # print("Legax=if" + self.legax)
-
-            # If axes exist, adjust their position.
-            else:
-                self.legax.set_position([1 - ax_extent_x, 0.01, ax_extent_x, ax_extent_y])
-                # self.legax.set_yticklabels("ax2", fontsize=32, minor=False)
-                # print("Legax=else" + self.legax)
-            # Plot the new legimg in the legax axes.
-            self.legimg = self.legax.imshow(img, origin=PIL_IMAGE_ORIGIN, aspect="equal", interpolation="nearest")
-            # print("Legimg" + self.legimg)
-        self.draw()
```

Comment:
I do not know where that "draw" is defined. It belongs to the "Canvas" and *this* object here *is* a canvas. So maybe it belongs just here? I think I (re)moved at least one of these.
Please test out, if it belongs here as self.draw(). You replaced it above with "self.ax.figure.canvas.draw()" which might be correct, but seems rather complicated